app/dojo.py

```diff
@@ -0,0 +1,291 @@
+            available_livingspace.available_capacity -= 1
+
+            cprint("\n %s will reside in %s\n" 
+                % (str(person), str(available_livingspace)), "green")
+        elif person.accomodation == "Pending":
+            person.accomodation = "Unallocated"
+            cprint("\n %s has not been allocated to"
+                 "a living space" % (str(person)), "magenta")
+        else:
+            pass
+        
+    def check_room_availability(self, category):
+        """Finds and return a random room that's not yet full"""
+        try:
+            available_room = choice(
+                    [room for room in self.rooms
+                     if room.category == category and room.available_capacity])
+        except IndexError:
+            return
+        return available_room
+
+    def add_person(self, category, name, surname, accomodation="N"):
+        name = str(name).capitalize() + " " + str(surname).capitalize()
+        accomodation = str(accomodation).capitalize() if accomodation else "N"
+        category = str(category).capitalize()
+        person_is_invalid = self.check_person_invalidity(name, accomodation, category)
+        
+        if person_is_invalid:
+            cprint(person_is_invalid, "magenta")
+            return person_is_invalid
+        
+        office = "Pending"
+        accomodation = "Pending" if accomodation == "Y" and category == "Fellow" else ""
+        
+        if category == "Fellow":
+            person = Fellow(name, office, accomodation)
+        else:
+            person = Staff(name, office)
+        
+        person.idno = str(len(self.persons) + 1)
+        self.persons.append(person)
+        cprint("\n %s has been added to the Dojo as a %s" % (name, category), "green")
+        self.allocate_office(person)
+        if accomodation == "Pending":
+            self.allocate_livingspace(person)
+
+    def check_person_invalidity(self, name, accomodation, category):
+        """Checks whether a new person has invalid arguments and 
+        returns an explantion"""
+        name_split = name.split(" ")        
+        if not name_split[0].isalpha() or not name_split[1].isalpha():
+            return ("\n %s contains non-alphabets\n"
+                "Person not added\n" % name)
+        if self.get_person_by_attribute(name, "name") != "Invalid name":
+            return ("\n %s already exists\n"
+            "Person not added\n" % name)
+        if accomodation not in ("N","Y"):
+            return ("Choose either 'Y' or 'N' for accomodation")
+        if category not in ("Fellow", "Staff"):
+            return "Invalid person category"
+        
+    def get_person_by_attribute(self, var, attr):
+        """This function receives two argument the attribute(attr - eg "name") and the
+        specific attribute of the person eg "George Bush" It returns the person object
+        It allows for flexibility since a one can request with any valid unique attr such as 
```

Comment:
great idea having this method separated.  Just some typo error in your comment statement